base/float.jl

```diff
@@ -1,5 +1,7 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+const IEEEFloat = Union{Float16, Float32, Float64}
```

Comment:
I realise that it is beyond the scope of this PR, but it would be nice if this were an abstract type so I could reuse all this for `Float128` once #22649 is fixed.